### ISSUE TITLE
Stockton-on-Tees, UK - Calendar Blank

### ISSUE THREAD

**Issue by geecy84** [mampfes/hacs_waste_collection_schedule#1171]
UK, Stockton-on-Tees Borough Council
Calendar does not pull any data, calendar appears blank.

```
waste_collection_schedule:
  sources:
    - name: stockton_gov_uk
      args:
        uprn: uprninteger
```

**Comment by 5ila5** [comment]
There was an Error with regex. It filtered out the st of august which lead to an Exception while parsing the date.

Fixed by #1174 Will be part of release 1.42.0 or is available now as master/git version

Will probably work in September again without updating

**Comment by geecy84** [comment]
The same thing has happened again. It was working after you fixed it last time, then this month it’s all gone again with no waste collections showing of any type.

**Comment by 5ila5** [comment]
the test returns valid data:

```bash
➜ ./custom_components/waste_collection_schedule/waste_collection_schedule/test/test_sources.py -s stockton_gov_uk -lit
Testing source stockton_gov_uk ...
  found 3 entries for 100110203615
    2023-11-14 : Waste [mdi:trash-can]
    2023-11-14 : Recycling [mdi:recycle]
    2023-10-31 : Garden Waste [mdi:leaf]
  found 3 entries for 100110160417
    2023-11-17 : Waste [mdi:trash-can]
    2023-11-24 : Recycling [mdi:recycle]
    2023-10-27 : Garden Waste [mdi:leaf]
  found 3 entries for 20002027430
    2023-11-17 : Waste [mdi:trash-can]
    2023-11-24 : Recycling [mdi:recycle]
    2023-10-27 : Garden Waste [mdi:leaf]
  found 3 entries for 100110206644
    2023-11-16 : Waste [mdi:trash-can]
    2023-11-16 : Recycling [mdi:recycle]
    2023-11-02 : Garden Waste [mdi:leaf]
```

Could you try one of these tests? Can you check the website if your address return collection dates in different format?

**Comment by geecy84** [comment]
In my logs I have: 

2023-11-16 20:29:33.833 ERROR (SyncWorker_30) [waste_collection_schedule.source_shell] fetch failed for source Stockton-on-Tees Borough Council:
Traceback (most recent call last):
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 134, in fetch
    entries = self._source.fetch()
              ^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/stockton_gov_uk.py", line 115, in fetch
    date = datetime.strptime(date_string, "%a %d %B %Y").date()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/_strptime.py", line 568, in _strptime_datetime
    tt, fraction, gmtoff_fraction = _strptime(data_string, format)
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/_strptime.py", line 349, in _strptime
    raise ValueError("time data %r does not match format %r" %
ValueError: time data 'Date not available' does not match format '%a %d %B %Y'

2023-11-16 20:30:14.125 ERROR (SyncWorker_17) [waste_collection_schedule.source_shell] fetch failed for source Stockton-on-Tees Borough Council:
Traceback (most recent call last):
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 134, in fetch
    entries = self._source.fetch()
              ^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/stockton_gov_uk.py", line 115, in fetch
    date = datetime.strptime(date_string, "%a %d %B %Y").date()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/_strptime.py", line 568, in _strptime_datetime
    tt, fraction, gmtoff_fraction = _strptime(data_string, format)
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/_strptime.py", line 349, in _strptime
    raise ValueError("time data %r does not match format %r" %
ValueError: time data 'Date not available' does not match format '%a %d %B %Y'

2023-11-16 20:30:17.523 ERROR (SyncWorker_9) [waste_collection_schedule.source_shell] fetch failed for source Stockton-on-Tees Borough Council:
Traceback (most recent call last):
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 134, in fetch
    entries = self._source.fetch()
              ^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/stockton_gov_uk.py", line 115, in fetch
    date = datetime.strptime(date_string, "%a %d %B %Y").date()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/_strptime.py", line 568, in _strptime_datetime
    tt, fraction, gmtoff_fraction = _strptime(data_string, format)
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/_strptime.py", line 349, in _strptime
    raise ValueError("time data %r does not match format %r" %
ValueError: time data 'Date not available' does not match format '%a %d %B %Y'

2023-11-16 20:30:27.807 ERROR (SyncWorker_21) [waste_collection_schedule.source_shell] fetch failed for source Stockton-on-Tees Borough Council:
Traceback (most recent call last):
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 134, in fetch
    entries = self._source.fetch()
              ^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/stockton_gov_uk.py", line 115, in fetch
    date = datetime.strptime(date_string, "%a %d %B %Y").date()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/_strptime.py", line 568, in _strptime_datetime
    tt, fraction, gmtoff_fraction = _strptime(data_string, format)
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/_strptime.py", line 349, in _strptime
    raise ValueError("time data %r does not match format %r" %
ValueError: time data 'Date not available' does not match format '%a %d %B %Y'

**Comment by geecy84** [comment]
Well it had come back on and was showing again, and now it's gone again and I have no entries. I don't know how to do the tests :( It's such a great integration, its really handy to have the collections shown, but I seem to be having no luck and it's hit and miss.

**Comment by 5ila5** [comment]
Sorry did not realize but this was allready fixed by #1375. 
The fix will be part of release 1.44.0 or is available now as master (git) version (HACS: 3-dot menu -> redownload -> change version to master)
